expression/builtin_miscellaneous.go

```diff
@@ -1044,3 +1048,148 @@ type uuidShortFunctionClass struct {
 func (c *uuidShortFunctionClass) getFunction(ctx sessionctx.Context, args []Expression) (builtinFunc, error) {
 	return nil, errFunctionNotExists.GenWithStackByArgs("FUNCTION", "UUID_SHORT")
 }
+
+type uuidToBinFunctionClass struct {
+	baseFunctionClass
+}
+
+func (c *uuidToBinFunctionClass) getFunction(ctx sessionctx.Context, args []Expression) (builtinFunc, error) {
+	if err := c.verifyArgs(args); err != nil {
+		return nil, err
+	}
+	argTps := []types.EvalType{types.ETString}
+	if len(args) == 2 {
+		argTps = append(argTps, types.ETInt)
+	}
+	bf, err := newBaseBuiltinFuncWithTp(ctx, c.funcName, args, types.ETString, argTps...)
+	if err != nil {
+		return nil, err
+	}
+
+	bf.tp.Flen = 16
+	types.SetBinChsClnFlag(bf.tp)
+	bf.tp.Decimal = 0
+	sig := &builtinUUIDToBinSig{bf}
+	return sig, nil
+}
+
+type builtinUUIDToBinSig struct {
+	baseBuiltinFunc
+}
+
+func (b *builtinUUIDToBinSig) Clone() builtinFunc {
+	newSig := &builtinUUIDToBinSig{}
+	newSig.cloneFrom(&b.baseBuiltinFunc)
+	return newSig
+}
+
+// evalString evals UUID_TO_BIN(string_uuid, swap_flag).
+// See https://dev.mysql.com/doc/refman/8.0/en/miscellaneous-functions.html#function_uuid-to-bin
+func (b *builtinUUIDToBinSig) evalString(row chunk.Row) (string, bool, error) {
+	val, isNull, err := b.args[0].EvalString(b.ctx, row)
+	if isNull || err != nil {
+		return "", isNull, err
+	}
+
+	u, err := uuid.Parse(val)
+	if err != nil {
+		return "", false, errWrongValueForType.GenWithStackByArgs("string", val, "uuid_to_bin")
+	}
+	bin, err := u.MarshalBinary()
+	if err != nil {
+		return "", false, errWrongValueForType.GenWithStackByArgs("string", val, "uuid_to_bin")
+	}
+
+	flag := int64(0)
+	if len(b.args) == 2 {
+		flag, isNull, err = b.args[1].EvalInt(b.ctx, row)
+		if isNull {
+			flag = 0
+		}
+		if err != nil {
+			return "", false, err
+		}
+	}
+	if flag != 0 {
+		swapBin := make([]byte, len(bin))
+		copy(swapBin[0:2], bin[6:8])
+		copy(swapBin[2:4], bin[4:6])
+		copy(swapBin[4:8], bin[0:4])
+		copy(swapBin[8:], bin[8:])
+		return string(swapBin), false, nil
```

Comment:
@b41sh sorry for our late reply.
Can we modify `bin` here? Then we can:
```
buf := make([]byte, 4)
copy(bin, bin[0:4]
copy(bin[0:2], bin[6:8]
copy(bin[2:4], bin[4:6];
copy(bin[4:8], but)
return string(bin)
```

Reply:
And how about creating a function for this logic? Then both the row-based and vectorized implementation can use it.

Reply:
Almost LGTM, please address comments. @b41sh

Reply:
I add two functions `swapBinaryUUID` and `swapStringUUID` to handle UUID swapping
PTAL @qw4990